docs/usage-guide.md

```diff
@@ -74,6 +73,140 @@ In addition to the arguments [mentioned above](#all-questions), `Choice` also ac
 `Confirm` doesn't take any additional arguments that weren't [mentioned above](#all-questions). However, the `default`
 argument takes a `bool` instead of `str`.
 
+## Only Asking Some Questions
+
+There are situations where a question should be asked some times, but not all the time. A program that is collects
+information about a user's pets should not ask the user for the dog's name and breed if the user said they do not have a
+dog. The `should_ask` argument that is present on each question provides a way to achieve this functionality.
+
+Similarly, `should_ask` can be used to provide branching paths to the user. An example of these branching paths is a
+[Choose Your Own Adventure][choose-your-own-adventure] story. The story provides the reader with choices during the
+adventure. These choices introduce diverging paths of interactions that may or may not join at the end.
+
+!!! warning Be careful when skipping a question
+    When `columbo` skips over a question, the `Answers` dictionary will **NOT** contain a key-value pair for the
+    skipped question. The examples below will demonstrate this result.
+
+### Optional Questions
+
+The following is a basic example that has two optional questions that are not asked based on the answer to the first
+question.
+
+```python
+import columbo
+
+def does_user_have_a_dog(answers: columbo.Answers) -> bool:
+    return answers["has_dog"]
+
+interactions = [
+    columbo.Confirm("has_dog", "Do you have a dog?", default=True),
+    columbo.BasicQuestion(
+        "dog_name",
+        "What is the name of the dog?",
+        should_ask=does_user_have_a_dog,
+        default="Kaylee"
+    ),
+    columbo.BasicQuestion(
+        "dog_breed",
+        "What is the breed of the dog?",
+        should_ask=does_user_have_a_dog,
+        default="Basset Hound"
+    )
+]
+
+user_answers = columbo.get_answers(interactions)
+print(user_answers)
+```
+
+If the user accepts the default answers for each of these questions, the output will be.
+
+```
+{"has_dog": True, "dog_name": "Kaylee", "dog_breed": "Basset Hound"}
+```
+
+However, when the user answers the first question with no, the output will be.
+
+```
+{"has_dog": False}
+```
+
+Note that in this case, the `Answers` dictionary only has an answer to the first question.
+
+### Branching Paths
+
+A question that is part of a branching path very similar to an optional question. It is still a question where
+`should_ask` was given to determine if the question should be skipped our not. The branching aspect comes there being at
+least two sets of optional questions. Each set has a `should_ask` argument that checks for a different state for a
+single answer. In this way, only one of the sets of optional questions will ever be asked.
+
+The following is an example of a short story that has two divergent paths that join at the end. Each individual question
+isn't different from the optional questions [demonstrated above](#optional-questions). The program achieves the
```

Comment:
```suggestion
The following is an example of a short story that has two divergent paths that join at the end. Each individual question
is an example of the optional questions [demonstrated above](#optional-questions). The program achieves the
```

Reply:
What I was trying to convey here is that questions used for branching are optional questions used in a specific way. I think the suggested change moves the sentence away from conveying that information. Do you have an idea that would better describe that aspect of the concept?